grapher/detailsOnDemand/detailsOnDemand.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react"
+import { Tippy } from "../chart/Tippy.js"
+import { MarkdownTextWrap } from "../text/MarkdownTextWrap.js"
+import { computed, observable, ObservableMap } from "mobx"
+import { Detail } from "../../clientUtils/owidTypes.js"
+import { observer } from "mobx-react"
+
+class DetailsOnDemand {
+    @observable details = new ObservableMap<
+        string,
+        ObservableMap<string, Detail>
+    >()
+
+    @observable getDetail(category: string, term: string): Detail | void {
+        const terms = this.details.get(category)
+        if (terms) {
+            const detail = terms.get(term)
+            return detail
+        }
+    }
+
+    @observable addDetails(details: Record<string, Record<string, Detail>>) {
+        const observableDetails = Object.entries(details).reduce(
+            (acc, [category, terms]) => {
+                acc.set(category, new ObservableMap(terms))
+                return acc
+            },
+            new ObservableMap()
+        )
+        this.details = this.details.merge(observableDetails)
+    }
+}
+
+// An object that can be shared throughout the codebase to make details globally accessible
+export const globalDetailsOnDemand = new DetailsOnDemand()
```

Comment:
Not sure about this pattern, was just trying to set something up for if/when we start having DoDs in articles.

It also makes it much easier for the MarkdownTextWrap to hook into